apps/core/components/searchForm/SearchFormModes.js

```diff
@@ -75,6 +95,25 @@ class SearchFormModes extends React.Component<Props> {
           infants={infants}
           bags={bags}
         />
+        <Modal isVisible={this.state.isModalOpen} onClose={this.onClose}>
+          {this.state.formMode === FORM_MODE.TRIP_TYPE && (
+            <Select
+              optionsData={TRIP_TYPE}
+              selectedType={tripType}
+              onSelect={this.handleTripTypeSelect}
+              onClosePress={this.onClose}
+            />
+          )}
+          {this.state.formMode === FORM_MODE.PASSENGERS && (
```

Comment:
I think this component is getting little bit bigger and obviously dealing with two different parts.
What about splitting it into separate `TripType` a `Passengers` related components? Each with its own logic and modal implementation?

Reply:
I see that Trond already suggested this one ⬆️. So I'll resolve this duplicate comment.